src/statcache.c

```diff
@@ -700,16 +777,21 @@ void stat_cache_delete_older(stat_cache_t *cache, const char *path_prefix, unsig
     log_print(LOG_DEBUG, SECTION_STATCACHE_CACHE, "stat_cache_delete_older: %s", path_prefix);
     iter = stat_cache_iter_init(cache, path_prefix);
     while ((entry = stat_cache_iter_current(iter))) {
-        log_print(LOG_DEBUG, SECTION_STATCACHE_CACHE, "stat_cache_delete_older: %s: min_gen %lu: loc_gen %lu",
-            entry->key, minimum_local_generation, entry->value->local_generation);
-        if (entry->value->local_generation < minimum_local_generation) {
-            stat_cache_delete(cache, key2path(entry->key), &tmpgerr);
-            ++deleted_entries;
-            if (tmpgerr) {
-                g_propagate_prefixed_error(gerr, tmpgerr, "stat_cache_delete_older: ");
-                free(entry);
-                stat_cache_iterator_free(iter);
-                return;
+        // Ignore negative (non-existent) entries, those tagged with st_mode == 0
+        // REVIEW: Do we want to leave negative entries, since we're doing a delete,
```

Comment:
Still cogitating on this. I think we don't want to remove negative entries when this function prunes positive entries which have been deleted. And we might just change positive entries to negative ones rather than call stat_cache_delete. Not sure.